pkg/datastructures/hashtable/main.go

```diff
@@ -0,0 +1,55 @@
+package hashtable
+
+type (
+	item struct {
+		next  *item
+		key   string
+		value string
+	}
+	bucket struct {
+		node   *item
+		length int
+	}
+)
+
+var table = make([]*bucket, 100)
+
+// hash function, adds up the rune values of the characters in the key
+func hash(key string) int {
```

Comment:
Though the algorithm is CORRECT, a better hashing function could be used to avoid obvious collisions, eg: *abcdef*, *bcdefa*, *cdefab*, and *defabc* all produce the same hash.
 A simple enhancement could be adding the **positional multiplier** to the hash function, it will greatly reduce the collisions.

Reply:
I was relying on the linked buckets items to handle collisions, changing the function is a much better solution.

Reply:
Yup, because if there are more collision then there will be more nodes in the bucket and more comparisons will be made in the linked list to find the entry

More Collisions --> Bigger Buckets --> More comparisons to find node in a bucket